### DISCUSSION TITLE
About load custom Scirpt

### DISCUSSION THREAD

**Discussion by Lafatalite** [pyri-project/pyri-core#15] [Q&A]
Hello,
I want to load three.js for custom development on a front-end page. I noticed that directly inserting the <script> tag into the HTML file within the Vue framework doesn't work. What should I do to load a custom script correctly?

**Comment by johnwason** [comment]
Take a look at the motion program package using the `load_js_src()` function: https://github.com/pyri-project/pyri-robotics-motion-program-browser/blob/master/src/pyri/robotics_motion_program_browser/robotics_mp_plugin_init.py

This package also uses a web server plugin to add resource files: https://github.com/pyri-project/pyri-robotics-motion-program/blob/e896554301ddc25b8bebde2b5610496fda4c7b32/src/pyri/robotics_motion_program/webui_server.py#L9

The `PyriWebUIResourceRouteHandler(__package__).handler` will automatically serve the contents of `webui_static` in the package.

**Reply by Lafatalite** [reply]
Hello,I am trying to add a page that needs to display several images. I attempted to add a `webui_static` folder to the project and inject it using `PyriWebUIResourceRouteHandler(__package__).handler`, but it was unsuccessful. I added a `webui_server.py` file and then added a load point in `pyproject.toml,` but I cannot access the images in `webui_static` from the frontend. Where did I go wrong?
![image](https://github.com/user-attachments/assets/58c6d7c8-e50f-464b-ae5f-dae414c00652)
![image](https://github.com/user-attachments/assets/8bb5fbc5-9a2e-45bd-9c43-a8fdc8697ded)
![image](https://github.com/user-attachments/assets/a176cd30-cebf-459e-87c6-eb51d5be1c33)

**Reply by johnwason** [reply]
Hi @Lafatalite 

The simplest solution to showing images is to use "data urls" and embed the image as base64. https://developer.mozilla.org/en-US/docs/Web/URI/Schemes/data

I will need to track down the problem you are having with the webui server plugin but I may not have time for a few days.
